command.go

```
@@ -5484,6 +5484,10 @@ func (cmd *InfoCmd) String() string {
 	return cmdString(cmd, cmd.val)
 }
 
+var (
+	moduleRe = regexp.MustCompile(`module:name=(.+?),(.+)$`)
+)
+
```

Hello @alingse thank you for the contribution. You are right that this may increase performance (if info modules is used often) and also will fail early if the regex is incorrect. At the same time time it introduces a global variable that can be changed from anywhere in the package. Since this method is not expected to be called that often I do not see the benefit outweighing the possible issues this may introduce. What I can suggest is to compile the regex only when used (i.e. if the section is modules) which will speed things a little.

---

😂  Yes, I also question the necessity of such changes. You can review the all results in the issue 7 of all top 2500 projects—there are too many instances of this usage, and most of them probably don't require the proposed optimization 🤔, How do you think

---

I cannot speak for the 2500 projects you have tagged, but for this one - I do not think it is needed. I would assume is a case by case optimization that can benefit high-traffic flows. If you would like to improve this repository, take a brief look at the function. The `MustCompile` can be moved within the if and it will improve the performance in theory without exposing a global variable. Feel free to add that change or close this PR.

---

🤔，ok，I will update it later, move the `MustCompile` within the if branch

---

Hi @alingse, I think you move `moduleRe = regexp.MustCompile` inside loop will make func more expensive?